dlt/common/runners/stdout.py

```diff
@@ -24,6 +25,45 @@ def exec_to_stdout(f: AnyFun) -> Iterator[Any]:
             print(encode_obj(rv), flush=True)
 
 
+def iter_std(venv: Venv, command: str, *script_args: Any) -> Iterator[Tuple[int, str]]:
+    """Starts a process `command` with `script_args` in environment `venv` and returns iterator
+    of (filno, line) tuples where `fileno` is 1 for stdout and 2 for stderr. `line` is
+    a content of a line with stripped new line character.
+
+    Use -u in scripts_args for unbuffered python execution
+    """
+    with venv.start_command(
+        command, *script_args, stdout=PIPE, stderr=PIPE, bufsize=1, text=True
+    ) as process:
+        exit_code: int = None
+        q_: queue.Queue[Tuple[int, str]] = queue.Queue()
+
+        def _r_q(std_: int) -> None:
+            stream_ = process.stderr if std_ == 2 else process.stdout
+            for line in iter(stream_.readline, ""):
+                q_.put((std_, line[:-1] if line[-1] == "\n" else line))
```

Comment:
I think `line.strip()` is simpler.

Reply:
Also can line in theory be `None`?

Reply:
I think None stops the iterator so no. I could not find it in my tests